src/frontend/js/types/Joanie.ts

```diff
@@ -101,16 +89,25 @@ export interface Product {
   target_courses: TargetCourse[];
   created_on: string;
   remaining_order_count?: number | null;
+  state: CourseState;
 }
 
 export interface CourseProduct extends Product {
   order: Nullable<OrderLite>;
   target_courses: TargetCourse[];
 }
 
+export interface CourseProductRelationCourse {
+  id: string;
+  code: string;
+  title: string;
+  cover?: JoanieFile;
+}
```

Comment:
In order to improve maintainability, what do you think to herit this type from `AbstractCourse`

```suggestion
export type CourseProductRelationCourse = Pick<AbstractCourse, 'id' | 'code' | 'title' | 'cover'>
```